packages/lib/modules/pool/actions/LiquidityActionHelpers.ts

```diff
@@ -61,12 +64,44 @@ export class LiquidityActionHelpers {
   }
 
   /* Used by default nested SDK handlers */
-  public get nestedPoolState(): NestedPoolState {
+  public get nestedPoolStateV2(): NestedPoolState {
     const result = mapPoolToNestedPoolStateV2(this.pool as PoolGetPool)
     result.protocolVersion = 2
     return result
   }
 
+  /* Used by default nested SDK handlers */
+  public get nestedPoolStateV3(): NestedPoolState {
+    const result = mapPoolToNestedPoolStateV3(this.pool as PoolGetPool)
+    result.protocolVersion = 3
+    return result
+  }
+
+  /* Used by V3 boosted SDK handlers */
+  public get boostedPoolState(): PoolStateWithUnderlyings & { totalShares: HumanAmount } {
+    const poolTokensWithUnderlyings: PoolTokenWithUnderlying[] = this.pool.poolTokens.map(
+      (token, index) => ({
+        ...token,
+        address: token.address as Address,
+        underlyingToken: {
+          ...token.underlyingToken,
+          address: token.underlyingToken?.address as Address,
+          decimals: token.underlyingToken?.decimals as number,
+          index, //TODO: review that this index is always the expected one
```

Comment:
@gareth The reference pool works but I have to double check this with the SDK team.